### PR TITLE
Detect M1 mac in test and substitute container image (#57)

### PR DESCRIPTION
**Pull request by alpreu**
This is a best-effort solution for issue #57.

It uses the JVMs system properties to detect the M1 architecture.
This solution will fail if the JVM is running through Rosetta emulation. Other approaches that are independent of the JVM suffer similar problems (see [this question about detecting the M1 from the cli](https://stackoverflow.com/questions/65259300/detect-apple-silicon-from-command-line) for example) so I believe there is no really good solution for this edge case.

### CODE REVIEW

File: spring-pulsar-spring-boot-autoconfigure/src/test/java/org/springframework/pulsar/autoconfigure/AbstractContainerBaseTests.java [hunk 1]
```diff
@@ -38,4 +39,18 @@ protected static String getHttpServiceUrl() {
 		return PULSAR_CONTAINER.getHttpServiceUrl();
 	}
 
+	private static boolean isRunningOnMacM1() {
```
Comment (onobc) [inline spring-pulsar-spring-boot-autoconfigure/src/test/java/org/springframework/pulsar/autoconfigure/AbstractContainerBaseTests.java:42]: > It uses the JVMs system properties to detect the M1 architecture.
This solution will fail if the JVM is running through Rosetta emulation. Other approaches that are independent of the JVM suffer similar problems (see [this question about detecting the M1 from the cli](https://stackoverflow.com/questions/65259300/detect-apple-silicon-from-command-line) for example) so I believe there is no really good solution for this edge case.

I think this is a pragmatic choice. We could later provide an env var that users in the "rosetta" group could set as an override/last resort. Again, I think this is fine for now. 


If we start needing to do this same logic in other places we could add something like `MacM1Detector` that encapsulates how we determine this info.

**Review by onobc — COMMENTED** [review]
Looks good @alpreu !
